src/backend/distributed/shardsplit/shardsplit_shared_memory.c

```diff
@@ -0,0 +1,389 @@
+									  dsm_handle *dsmHandle)
+{
+	if (shardSplitInfoCount <= 0 || shardSplitInfoSize <= 0)
+	{
+		ereport(ERROR,
+				(errmsg("shardSplitInfoCount and size of each step should be "
+						"positive values")));
+	}
+
+	Size totalSize = offsetof(ShardSplitInfoSMHeader, splitInfoArray) +
+					 (shardSplitInfoCount * shardSplitInfoSize);
+	dsm_segment *dsmSegment = dsm_create(totalSize, DSM_CREATE_NULL_IF_MAXSEGMENTS);
+
+	if (dsmSegment == NULL)
+	{
+		ereport(ERROR,
+				(errmsg("could not create a dynamic shared memory segment to "
+						"store shard split info")));
+	}
+
+	*dsmHandle = dsm_segment_handle(dsmSegment);
+
+	/*
+	 * Pin the segment till Postmaster shutsdown since we need this
+	 * segment even after the session ends for replication catchup phase.
+	 */
+	dsm_pin_segment(dsmSegment);
+
+	ShardSplitInfoSMHeader *shardSplitInfoSMHeader =
+		GetShardSplitInfoSMHeaderFromDSMHandle(*dsmHandle);
+
+	shardSplitInfoSMHeader->count = shardSplitInfoCount;
+
+	return shardSplitInfoSMHeader;
+}
+
+
+/*
+ * CreateSharedMemoryForShardSplitInfo is a wrapper function which creates shared memory
+ * for storing shard split infomation. The function returns pointer to the header of
+ * shared memory segment.
+ *
+ * shardSplitInfoCount - number of 'ShardSplitInfo ' elements to be allocated
+ * dsmHandle           - handle of the allocated shared memory segment
+ */
+ShardSplitInfoSMHeader *
+CreateSharedMemoryForShardSplitInfo(int shardSplitInfoCount, dsm_handle *dsmHandle)
+{
+	ShardSplitInfoSMHeader *shardSplitInfoSMHeader =
+		AllocateSharedMemoryForShardSplitInfo(shardSplitInfoCount,
+											  sizeof(ShardSplitInfo),
+											  dsmHandle);
+	return shardSplitInfoSMHeader;
+}
+
+
+/*
+ * ReleaseSharedMemoryOfShardSplitInfo releases(unpins) the dynamic shared memory segment
+ * allocated by 'worker_split_shard_replication_setup'. This shared memory was pinned
+ * to Postmaster process and is valid till Postmaster shutsdown or
+ * explicitly unpinned by calling 'dsm_unpin_segment'.
+ */
+void
+ReleaseSharedMemoryOfShardSplitInfo()
+{
+	/* Get handle of dynamic shared memory segment*/
+	dsm_handle dsmHandle = GetShardSplitSharedMemoryHandle();
+
+	/*
+	 * Unpin the dynamic shared memory segment. 'dsm_pin_segment' was
+	 * called previously by 'AllocateSharedMemoryForShardSplitInfo'.
+	 */
+	dsm_unpin_segment(dsmHandle);
+
+	/*
+	 * As dynamic shared memory is unpinned, store an invalid handle in static
+	 * shared memory used for handle management.
+	 */
+	StoreShardSplitSharedMemoryHandle(DSM_HANDLE_INVALID);
+}
+
+
+/*
+ * encode_replication_slot returns an encoded replication slot name
+ * in the following format.
+ * Slot Name = citus_split_nodeId_tableOwnerOid
+ * Max supported length of replication slot name is 64 bytes.
+ */
+char *
+encode_replication_slot(uint32_t nodeId,
+						uint32_t tableOwnerId)
+{
+	StringInfo slotName = makeStringInfo();
+	appendStringInfo(slotName, "%s%u_%u", SHARD_SPLIT_REPLICATION_SLOT_PREFIX, nodeId,
```

Comment:
nitpick: Instead of %s%u_%u, %s_%u_%u feels more cleaner.

Reply:
This pattern is followed from existing MOVE operations. The X_PREFIX macros have "_" as suffix. Continued this.